zkevm-circuits/src/evm_circuit/util/math_gadget.rs

```diff
@@ -778,21 +778,15 @@ pub(crate) fn generate_lagrange_base_polynomial<
 /// MulAddWordsGadget.
 #[derive(Clone, Debug)]
 pub(crate) struct MulAddWordsGadget<F> {
-    pub a: util::Word<F>,
-    pub b: util::Word<F>,
-    pub c: util::Word<F>,
-    pub d: util::Word<F>,
+    pub words: [util::Word<F>; 4],
```

Comment:
should we just keep the `words` in the `MulDivModGadget`?

Reply:
Agree. We don't need it to hold the words indeed. Just to get them via `configure/construct` fn

Reply:
I've moved the `words` out of `MulAddWordsGadget` and into `MulDivModGadget`. I've also changed the `construct` to receive a reference to the words instead.